app/controllers/actions/GetArrivalAction.scala

```diff
@@ -68,13 +69,19 @@ private[actions] class AuthenticatedGetArrivalAction(
     extends ActionRefiner[AuthenticatedRequest, ArrivalRequest] {
 
   override protected def refine[A](request: AuthenticatedRequest[A]): Future[Either[Result, ArrivalRequest[A]]] =
-    repository.get(arrivalId).map {
-      case Some(arrival) if arrival.eoriNumber == request.eoriNumber =>
-        Right(ArrivalRequest(request.request, arrival))
-      case Some(_) =>
-        Logger.warn("Attempt to retrieve an arrival for another EORI")
-        Left(NotFound)
-      case None =>
-        Left(NotFound)
-    }
+    repository
+      .get(arrivalId)
+      .map {
+        case Some(arrival) if arrival.eoriNumber == request.eoriNumber =>
+          Right(ArrivalRequest(request.request, arrival))
+        case Some(_) =>
+          Logger.warn("Attempt to retrieve an arrival for another EORI")
+          Left(NotFound)
+        case None =>
+          Left(NotFound)
+      }
+      .recover {
+        case e =>
+          Left(InternalServerError(s"Failed with the following error: $e"))
```

Comment:
I don't think we should return this error as part of the response but it makes sense to log it

Reply:
I'll log it as an error